src/Platforms/OraclePlatform.php

```diff
@@ -431,43 +432,65 @@ protected function _getCreateTableSQL($name, array $columns, array $options = []
         return $sql;
     }
 
+    public function getListDatabaseIndexesSQL(string $database): string
+    {
+        return $this->getListIndexesSQL($database);
+    }
+
     /**
      * {@inheritDoc}
      *
      * @link http://ezcomponents.org/docs/api/trunk/DatabaseSchema/ezcDbSchemaOracleReader.html
      */
     public function getListTableIndexesSQL($table, $database = null)
     {
-        $table = $this->normalizeIdentifier($table);
-        $table = $this->quoteStringLiteral($table->getName());
+        return $this->getListIndexesSQL($database, $table);
+    }
+
+    private function getListIndexesSQL(?string $database, ?string $table = null): string
+    {
+        $conditions = [];
 
-        return "SELECT uind_col.index_name AS name,
-                       (
-                           SELECT uind.index_type
-                           FROM   user_indexes uind
-                           WHERE  uind.index_name = uind_col.index_name
-                       ) AS type,
-                       decode(
-                           (
-                               SELECT uind.uniqueness
-                               FROM   user_indexes uind
-                               WHERE  uind.index_name = uind_col.index_name
-                           ),
-                           'NONUNIQUE',
-                           0,
-                           'UNIQUE',
-                           1
-                       ) AS is_unique,
-                       uind_col.column_name AS column_name,
-                       uind_col.column_position AS column_pos,
-                       (
-                           SELECT ucon.constraint_type
-                           FROM   user_constraints ucon
-                           WHERE  ucon.index_name = uind_col.index_name
-                       ) AS is_primary
-             FROM      user_ind_columns uind_col
-             WHERE     uind_col.table_name = " . $table . '
-             ORDER BY  uind_col.column_position ASC';
+        $sql = <<<SQL
+              SELECT ind_col.table_name,
+                     ind_col.index_name AS name,
+                     ind.index_type AS type,
+                     decode(ind.uniqueness, 'NONUNIQUE', 0, 'UNIQUE', 1) AS is_unique,
+                     ind_col.column_name AS column_name,
+                     ind_col.column_position AS column_pos,
+                     con.constraint_type AS is_primary
+SQL;
+
+        if (isset($database)) {
+            $sql         .= <<<SQL
+                FROM all_ind_columns ind_col
+           LEFT JOIN all_indexes ind ON ind.owner = ind_col.index_owner AND ind.index_name = ind_col.index_name
+           LEFT JOIN all_constraints con ON  con.owner = ind_col.index_owner AND con.index_name = ind_col.index_name
+SQL;
+            $conditions[] = 'ind_col.index_owner = ' . $this->quoteStringLiteral(
+                $this->normalizeIdentifier($database)->getName()
+            );
+            if (isset($table)) {
```

Comment:
```suggestion
            if ($table !== null) {
```
If you make a typo in the variable name, the `isset()` will swallow it.

Reply:
this was the only way to avoid Psalm complaining that `$table` may be null

Reply:
When I tried that, I got

```
ERROR: RedundantCondition - src/Platforms/OraclePlatform.php:454:13 - string can never contain null (see https://psalm.dev/122)
        if ($database !== null) {
```

Reply:
How is this error relevant to this code? The condition is about `$table` while the error is about `$database`.

Please do not use assertions to silence a static analyzer unless you're sure that the error message is invalid.